README.md

```diff
@@ -54,134 +55,62 @@ type Decorator = (value: Input, context: {
   isPrivate?: boolean;
   isStatic?: boolean;
   addInitializer?(initializer: () => void): void;
-+ metadataKey?: MetadataKey;
-+ class?: {
-+   metadataKey: MetadataKey;
-+   name: string;
-+ }
++ metadata?: Record<string | number | symbol, unknown>;
 }) => Output | void;
 ```
 
-Two new values are introduced, `metadataKey` and `class`.
+The new `metadata` property is a plain JavaScript object. The same object is
+passed to every decorator applied to a class or any of its elements. After the
+class has been fully defined, it is assigned to the `Symbol.prototype` property
+of the class.
 
-### `metadataKey`
-
-`metadataKey` is present for any _tangible_ decoratable value, specifically:
-
-- Classes
-- Class methods
-- Class accessors and auto-accessors
-
-It is not present for class fields because they have no tangible value (e.g.
-there is nothing to associate the metadata with, directly or indirectly).
-`metadataKey` is then set on the decorated value once decoration has completed:
+An example usage might look like:
 
 ```js
-const METADATA = new WeakMap();
-
-function meta(value) {
+function meta(key, value) {
   return (_, context) => {
-    METADATA.set(context.metadataKey, value);
+    context.metadata[key] = value;
```

Comment:
I'd suggest having both an example using the object itself, and one using the object only as a key to a `WeakMap`. I agree that package authors should probably use a `WeakMap` (or at least a `Symbol()`) to avoid collisions, but such collisions are far less likely for end-user applications/services where a single team owns the code, so both models are equally valid.

Reply:
Good point, added